public/offline.html

```diff
@@ -1,102 +1,124 @@
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
 
-    <base href="/"/>
-    
-    <link rel="apple-touch-icon" href="/logo192.png"/>
-    <link rel="manifest" href="/manifest.json"/>
-    <link href="/static/fonts/montserrat.css" type="text/css" rel="stylesheet" />
+    <base href="/" />
+
+    <link rel="apple-touch-icon" href="/logo192.png" />
+    <link rel="manifest" href="/manifest.json" />
+    <link
+      href="/static/fonts/montserrat.css"
+      type="text/css"
+      rel="stylesheet"
+    />
 
     <title>Zeeguu</title>
+  </head>
+  <style>
+    .contentDiv {
+      display: flex;
+      flex-direction: column;
+      align-items: center;
+      font-family: "Monsterrat", sans-serif;
+    }
 
-<style>
-  .contentDiv {
-    display: flex; 
-    flex-direction: column;
-    align-items: center;
-    font-family: 'Monsterrat', sans-serif;
-  }
+    .capitalizeBold {
+      text-transform: capitalize;
+      font-weight: bold;
+    }
+  </style>
 
-  .capitalizeBold {
-    text-transform: capitalize;
-    font-weight: bold;
-  }
-</style>
+  <body>
+    <div id="offlineRoot" class="contentDiv">
+      <img src="/logo192.png" />
+      <h1>You seem to be offline</h1>
+      <h2>Please return to Zeeguu when you are back online!</h2>
+      <div id="wordsList">Loading cached words...</div>
+    </div>
 
-<body>
-<div id="offlineRoot" class="contentDiv">
-  <img src="/logo192.png"> 
-  <h1>You seem to be offline</h1>
-  <h2>Please return to Zeeguu when you are back online!</h2>
-  <div id="wordsList">Loading cached words...</div>
-</div>
+    <script>
+      async function displayCachedWords() {
+        try {
+          const sessionId = document.cookie.split(";")[0].split("=")[1];
+          if (!sessionId) {
+            console.error("No session ID found");
+            return;
+          }
 
-<script>
-  async function displayCachedWords() {
-    try {
-      const sessionId = document.cookie.split(";")[0].split("=")[1]
-      if (!sessionId) {
-        console.error("No session ID found")
-        return;
-      }
-      
-      const cache = await caches.open("data-cache-v1");
-      const request = new Request(`https://api.zeeguu.org/bookmarks_in_pipeline?session=${sessionId}`);
-      const cachedRequests = await cache.keys();
-      let cachedResponse = null;
+          const cache = await caches.open("data-cache-v1");
+          const request = new Request(
+            `https://api.zeeguu.org/bookmarks_in_pipeline?session=${sessionId}`,
+          );
+          const cachedRequests = await cache.keys();
+          let cachedResponse = null;
 
-      for (const cachedRequest of cachedRequests) {
-        if (new URL(cachedRequest.url).pathname === new URL(request.url).pathname) {
-          cachedResponse = await cache.match(cachedRequest);
-          break;
-        }
-      }
+          for (const cachedRequest of cachedRequests) {
```

Comment:
I don't think you can have a const inside of a loop, as you won't be able to update the value (this will fail after one execution). 

Here is a small example:

![{AD553DCB-E31F-492E-9341-061C4713289C}](https://github.com/user-attachments/assets/41cffeb6-89a9-488b-9a8c-cfaf7539d163)

Reply:
I just looked into this, but it seems like there is a difference between the for loop I'm using and the one you showed. 

The difference seems to be that in my "for-of" loop, the `cachedRequest` variable is a new variable in each iteration, and thus the value of `cachedRequest` is never changed within the same iteration and no error should occur. 

Same method is used here, in the section about filtering:
https://web.dev/articles/cache-api-quick-guide#how_much_can_be_stored